### PR TITLE
Added some missing logic for some cuttelfish_datatype:to_string edge cases

### PR DESCRIPTION
**Pull request by joedevivo**
Fixes 

```
ERROR: generate failed while processing /home/buildbot/masters/riak/riak-build-centos-6-64/build/distdir/BUILD/riak-2.0.0pre9-6118bedf/rel: {'EXIT',
    {badarg,
        [{io_lib,format,
             ["Default: ~s",
              [{error,"could not convert off of type flag to a string"}]],
             [{file,"io_lib.erl"},{line,155}]},
         {cuttlefish_conf,generate_comments,1,
             [{file,"src/cuttlefish_conf.erl"},{line,125}]},
         {cuttlefish_conf,generate_element,1,
             [{file,"src/cuttlefish_conf.erl"},{line,108}]},
         {cuttlefish_conf,'-generate/1-fun-0-',2,
             [{file,"src/cuttlefish_conf.erl"},{line,71}]},
         {lists,foldl,3,[{file,"lists.erl"},{line,1248}]},
         {cuttlefish_conf,generate_file,2,
             [{file,"src/cuttlefish_conf.erl"},{line,76}]},
         {cuttlefish_rebar_plugin,make_default_file,3,
             [{file,"src/cuttlefish_rebar_plugin.erl"},{line,83}]},
         {cuttlefish_rebar_plugin,generate,2,
             [{file,"src/cuttlefish_rebar_plugin.erl"},{line,58}]}]}}
```

once that was fixed, this was exposed:

```
ERROR: generate failed while processing /Users/joe/dev/basho/riak_ee/rel: {'EXIT',
    {badarg,
        [{io_lib,format,
             ["Default: ~s",
              [{error,
                   "Tried to convert 1, an invalid datatype {integer,1} to_string."}]],
             [{file,"io_lib.erl"},{line,155}]},
         {cuttlefish_conf,generate_comments,1,
             [{file,"src/cuttlefish_conf.erl"},{line,126}]},
         {cuttlefish_conf,generate_element,1,
             [{file,"src/cuttlefish_conf.erl"},{line,108}]},
         {cuttlefish_conf,'-generate/1-fun-0-',2,
             [{file,"src/cuttlefish_conf.erl"},{line,71}]},
         {lists,foldl,3,[{file,"lists.erl"},{line,1248}]},
         {cuttlefish_conf,generate_file,2,
             [{file,"src/cuttlefish_conf.erl"},{line,76}]},
         {cuttlefish_rebar_plugin,make_default_file,3,
             [{file,"src/cuttlefish_rebar_plugin.erl"},{line,83}]},
         {cuttlefish_rebar_plugin,generate,2,
             [{file,"src/cuttlefish_rebar_plugin.erl"},{line,58}]}]}}
make: *** [generate] Error 1
```

Now both work.

### CODE REVIEW

**Comment by jaredmorrow** [comment]
Confirmed that 'make rel' works, did not do a proper code review though.

**Comment by seancribbs** [comment]
Can the unit test(s) be updated?

File: src/cuttlefish_datatypes.erl [hunk 1]
```diff
@@ -132,11 +150,20 @@ to_string(File, file) when is_list(File) -> File;
...
-to_string(Flag, {flag, _, _}=Type) -> cuttlefish_flag:to_string(Flag, Type);
+to_string(Flag, flag) when is_list(Flag) -> cuttlefish_flag:to_string(Flag, flag);
+to_string(Flag, {flag, _, _}=Type) when is_atom(Flag) -> cuttlefish_flag:to_string(Flag, Type);
+to_string(Flag, {flag, _, _}=Type) when is_list(Flag) -> cuttlefish_flag:to_string(Flag, Type);
 
 %% The Pokemon Clause: Gotta Catch 'em all!
-to_string(X, InvalidDatatype) ->
-    {error, lists:flatten(io_lib:format("Tried to convert ~p, an invalid datatype ~p to_string.", [X, InvalidDatatype]))}.
+to_string(Value, MaybeExtendedDatatype) ->
+    io:format("is_extended(~p) = ~p~n", [MaybeExtendedDatatype, is_extended(MaybeExtendedDatatype)]),
```
Comment (seancribbs) [inline src/cuttlefish_datatypes.erl]: Stray io:format.

**Comment by seancribbs** [comment]
:+1:

**Pull request merged**